tests/functional/test_install.py

```diff
@@ -749,7 +750,7 @@ def test_url_req_case_mismatch_no_index(script, data):
     tests/data/packages contains Upper-1.0.tar.gz and Upper-2.0.tar.gz
     'requiresupper' has install_requires = ['upper']
     """
-    Upper = os.path.join(data.find_links, 'Upper-1.0.tar.gz')
+    Upper = '/'.join((data.find_links, 'Upper-1.0.tar.gz'))
```

Comment:
Why not `os.path.join`?

Reply:
Because `data.find_links` is an URL.

Reply:
Oh. Okay. XD

Reply:
Maybe better to use `urljoin` then, just to be explicit? I wouldn't want someone in future to "helpfully" switch back to os.path.join (although the Windows tests would now pick that up :smile:)

Reply:
Except just replacing the example above with `urljoin(data.find_links, 'Upper-1.0.tar.gz')` does not work... Because:
```
>>> urljoin('/index/', 'package.tar.gz')
'/index/package.tar.gz'
>>> urljoin('/index', 'package.tar.gz
'/package.tar.gz'
```
Alternatively, use `from posixpath import join as urljoin`?